src/com/projectkorra/projectkorra/firebending/FireBlast.java

```diff
@@ -202,7 +216,24 @@ public void progress() {
 				furnace.setBurnTime((short) 800);
 				furnace.setCookTime((short) 800);
 				furnace.update();
-			} else if (BlazeArc.isIgnitable(this.player, block.getRelative(BlockFace.UP))) {
+			} else if (block.getType() == Material.SMOKER && this.powerFurnace) {
+				final Smoker smoker = (Smoker) block.getState();
+				smoker.setBurnTime((short) 800);
+				smoker.setCookTime((short) 800);
+				smoker.update();
+		    } else if (block.getType() == Material.BLAST_FURNACE && this.powerFurnace) {
+		    	final BlastFurnace blastF = (BlastFurnace) block.getState();
+				blastF.setBurnTime((short) 800);
+				blastF.setCookTime((short) 800);
+				blastF.update();
+		    } else if (block instanceof Campfire) {
+		    	final Campfire campfire = (Campfire) block.getBlockData();
+		    	if(!campfire.isLit()) {
+		    		if(block.getType() != Material.SOUL_CAMPFIRE || bPlayer.canUseSubElement(SubElement.BLUE_FIRE)) {

```

Comment:
Loving these changes to include new blocks. Nice job!

Reply:
Thank you!